### PR TITLE
Fix more warnings

### PR DESCRIPTION
**Pull request by adamdunkels**
Really close now, only one warning in the cc26xx submodule - probably need to involve @g-oikonomou to get that one fixed.

### CODE REVIEW

**Comment by g-oikonomou** [comment]
I flagged that particular one with TI before the most recent xxware release, but nothing happened...

I could of course fix it in xxware and then we can bump the submodule here

**Comment by adamdunkels** [comment]
Yeah, definitely fix it locally in your branch, then later hope that it gets merged upstream.

File: platform/srf06-cc26xx/contiki-conf.h [hunk 1]
```diff
@@ -334,6 +334,15 @@ typedef uint32_t rtimer_clock_t;
...
+
+/* The CC26xx port does not declare strcasecmp() and strncasecmp() in
+   its headers, so we need to declare them here for a warning-free
+   build. */
+#include <string.h>
+#include <stddef.h>
+int strcasecmp(const char *s1, const char *s2);
+int strncasecmp(const char *s1, const char *s2, size_t n);
+
 #endif /* CONTIKI_CONF_H */
```
Comment (g-oikonomou) [inline platform/srf06-cc26xx/contiki-conf.h]: I am not convinced by this one. If there some file somewhere that tries to use those functions, then the respective headers should be included there, no?

Comment (adamdunkels) [inline platform/srf06-cc26xx/contiki-conf.h]: Yes, they do, but the problem is that the string.h header for the cc26xx does not (for whatever reason) seem to have those two declarations.

Comment (adamdunkels) [inline platform/srf06-cc26xx/contiki-conf.h]: And the reason why string.h and stddef.h are included here is that they are needed for `size_t` to be defined.

Comment (g-oikonomou) [inline platform/srf06-cc26xx/contiki-conf.h]: Are you testing locally with arm-gcc 4.8.x?

If memory servers, under 4.8.x I had to include strings.h to use those, but with 4.9.x it is no longer required

**Comment by simonduq** [comment]
Great! Any reason not to have this in the common `Makefile.include`?
```
ifdef WERROR
CFLAGS += -Werror
endif
```

File: platform/srf06-cc26xx/contiki-conf.h (hunk 1 above)
Comment (adamdunkels) [inline platform/srf06-cc26xx/contiki-conf.h]: Oh, that would be perfect - I was indeed running 4.8.3 in my Instant Contiki. Great if this is not needed with 4.9!

**Comment by adamdunkels** [comment]
Yes, not all C compilers have the same nomenclature for this command line option (if they support the feature at all).

**Comment by simonduq** [comment]
OK. I'll add jn516x, and try to add as many platforms of compile-test then.
Do you need to update something before I merge based on the discussion with @g-oikonomou?

**Comment by adamdunkels** [comment]
Yes, give me a second, I'm testing with gcc 4.9.3 here!

**Comment by adamdunkels** [comment]
Let's wait for travis, then I'll squash a few of the commits, and then we should be ready to go.

**Comment by simonduq** [comment]
OK. Added WERROR to jn516x and also AVR (there are many AVR tests)

**Comment by adamdunkels** [comment]
I removed the cc26xx strncasecmp commits and tested this with AVR WERROR=1.

Unfortunately the AVR platforms seem need a larger overhaul before we can safely do a clean build. For example, the `clock_time_t` is too small, which causes overflows in the RPL timer code. So I'd suggest we'll hold off with that one for now.

**Comment by simonduq** [comment]
Good point, removed it on AVR for now

**Pull request merged**